Code/Framework/AzCore/AzCore/IO/Path/Path.inl

```diff
@@ -92,6 +93,16 @@ namespace AZ::IO::Internal
     constexpr auto ConsumeRootName(InputIt entryBeginIter, InputIt entryEndIter, const char preferredSeparator)
         -> AZStd::enable_if_t<AZStd::Internal::is_forward_iterator_v<InputIt>, InputIt>
     {
+    #if defined(AZ_TRAIT_CUSTOM_PATH_ROOT_SEPARATOR)
+        const AZStd::string_view path{ entryBeginIter, entryEndIter };
+        const auto positionOfPathSeparator = path.find(AZ_TRAIT_CUSTOM_PATH_ROOT_SEPARATOR);
+        if (positionOfPathSeparator == AZStd::string_view::npos)
+        {
+            return entryBeginIter;
+        }
+        const AZStd::string_view rootName{ path.substr(0, positionOfPathSeparator + 1) };
+        return AZStd::next(entryBeginIter, rootName.size());
+    #else
```

Comment:
This should be inside of the `if (preferredserparator == '/')` block.

Since this only applies to Posix Style paths.

Also can you update the if check to be `if (preferredSeparator == PosixPathSeparator)`

Reply:
Happy to make the change if you think it's best, but I figured it's better to leave it outside the if block in case it's ever needed by a non-posix platform?

Reply:
I think we we should make this change.

The Path API isn't a compile time API, so even on Windows a user could construct an `AZ:IO::Path posixPath(AZ::IO::PosixSeparator)` and the path will be treated as a Posix path.

If such a platform in the future which uses the Windows Path Separator, with a customized RootName ever exist, then the logic to support it can be implemented then.

Reply:
I was more thinking that the logic we're adding with this new trait is independent of the preferred separator, but doesn't matter much either way so I just updated it.